powerauth-nextstep/src/main/java/io/getlime/security/powerauth/app/nextstep/service/UserIdentityService.java

```diff
@@ -218,8 +208,9 @@ public CreateUserResponse createUserIdentity(CreateUserRequest request) throws U
             List<UserContactDetail> addedContacts = updateContacts(user, contacts);
             response.getContacts().addAll(addedContacts);
         }
-        // Save user identity snapshot to the history table
-        saveUserIdentityHistory(user);
+        // Save user identity and a snapshot to the history table
+        updateUserIdentityHistory(user);
```

Comment:
shouldn't it be reversed?
First userIdentity and then its History to be hooked on the userIdentity?

Reply:
The user identity history no longer uses a dedicated repository. So the history is updated (added to the user identity) and then it is saved by saving the user identity. The `updateUserIdentityHistory` method call does not call any `save()` method. So, the `save()` on userIdentity needs to be always called last, otherwise the changes wouldn't be saved into database.

Reply:
oh, you're right. Then it's good